### ISSUE TITLE
Cannot Host Co-Op Game

### ISSUE THREAD

**Issue by her001** [ValveSoftware/portal2#32]
When I try to host a co-op game on Linux, the person joining (from a windows machine) gets a non-responsive game, and I see a message saying that my partner has disconnected.

**Comment by nroach44** [comment]
Have you tried the other way around?

I know that hosting a PS3 player works.

**Comment by her001** [comment]
I was able to join the other person's game without any issues (besides latency, due to his internet).

**Comment by johndrinkwater** [comment]
FWIW, connected to another linux users’ game fine; also connected to same user with portal2 ps3.
Maybe just Win→Linux ?

**Comment by flamingspaz** [comment]
Can confirm when I host on Linux my friend on Windows gets put back into the menu with no buttons. The other way around works.

See http://steamcommunity.com/sharedfiles/filedetails/?id=232185805

**Comment by SleepyAkubi** [comment]
Can confirm.

**Comment by josefnpat** [comment]
+1 subscribing

**Comment by SuperTux88** [comment]
I have the same problem (Linux <-> Linux)

When the other person hosts the game, I see this in my console:
`
Connecting to public(192.168.2.38:27015) ...
446.735:  Sending Steam connect to 192.168.2.38:27015 1100001002cd60a
Initiate 1100001002cd60a
Generated 0.0.0.1:1 for 1100001002cd60a
Created NS_CLIENT connection to 1100001002cd60a
Retrying public(192.168.2.38:27015) ...`

192.168.2.38 is his LAN address, but we are not in the same LAN.

When I host the game, his game tries to connect to 127.0.1.1

It worked a few weeks ago.

**Comment by bndbsh** [comment]
Still an issue.

**Comment by LiamDawe** [comment]
Confirming this is an issue still years later.

Linux to Linux doesn't even work most of the time.

**Comment by VAlex0** [comment]
@LiamDawe @mninja @SuperTux88 @josefnpat @SleepyAkubi @flamingspaz @johndrinkwater @her001 

Hey guys. Im glad i can help here.
I had the same problem today with a friend of mine and could fix it.

tl;dr:
Use the ingame console to manually choose the friends ip

--------------

**My setup:**
Win 8.1 64 bit, WLAN Card, Asus Router, Portal 2 Steam version, the update from March 5 in 2018 (Today is 8th of April 2018)
I personally forwarded port 27015 UDP&TCP onto my PC, but i did not check yet if that is necessary

**My friends setup:**
Win 7 Pro 64bit, LAN Card, FritzBox Router, Same steam version of the game

**As it seems, the setup has no effect and there is a high chance that your setup can be fixxed too!**

-----------

**The problem:**
Portal 2 is gathering a wrong public ip. It is either the loopback (127.0.0.1) which happened on my system, or your own PC in your LAN, e.g. 192.168.1.42. This information is then sent to the other player via the steam game connect system.

--------------------

**The fix:**
* Manually connect to the host
* Manually start the coop map

For these commands you need to open the ingame console. Enable it in `settings -> keyboard/mouse -> Enable console`
Then you can use the console (default is key `\``)
1. The friend now needs to type into the console `connect 10.11.12.13` (=public ip of the host https://www.whatismyip.com/de/)
2. A little bit after that the host can initiate the coop map by typing in `map map mp_coop_lobby_2` This will create a coop game that starts in the lobby.
Important: The friend will try to recconect every 5 seconds for the end of the time. But the host will only load the map, and after 10 secs kick the host, because no other player joined! So best is to first connect the friend and then start the map


-----

Problem: cheats are turned on, so no achivements. To solve that type in `sv_cheats 0` and restart Portal 2 and the coop game (source: https://steamcommunity.com/app/620/discussions/0/846962626973701432/?ctp=2#c353915309344599851)

All map names in the right order:
The correct order of maps thanks to: Kpok 
```
"mp_coop_start" 
"mp_coop_lobby_2" 
"mp_coop_doors" 
"mp_coop_race_2" 
"mp_coop_laser_2" 
"mp_coop_rat_maze" 
"mp_coop_laser_crusher" 
"mp_coop_teambts" 
"mp_coop_fling_3" 
"mp_coop_infinifling_train" 
"mp_coop_come_along" 
"mp_coop_fling_1" 
"mp_coop_catapult_1" 
"mp_coop_multifling_1" 
"mp_coop_fling_crushers" 
"mp_coop_fan" 
"mp_coop_wall_intro" 
"mp_coop_wall_2" 
"mp_coop_catapult_wall_intro" 
"mp_coop_wall_block" 
"mp_coop_catapult_2" 
"mp_coop_turret_walls" 
"mp_coop_turret_ball" 
"mp_coop_wall_5" 
"mp_coop_tbeam_redirect" 
"mp_coop_tbeam_drill" 
"mp_coop_tbeam_catch_grind_1" 
"mp_coop_tbeam_laser_1" 
"mp_coop_tbeam_polarity" 
"mp_coop_tbeam_polarity2" 
"mp_coop_tbeam_polarity3" 
"mp_coop_tbeam_maze" 
"mp_coop_tbeam_end" 
"mp_coop_paint_come_along" 
"mp_coop_paint_redirect" 
"mp_coop_paint_bridge" 
"mp_coop_paint_walljumps" 
"mp_coop_paint_speed_fling" 
"mp_coop_paint_red_racer" 
"mp_coop_paint_speed_catch" 
"mp_coop_paint_longjump_intro" 
"mp_coop_credits"

```

**Comment by bigbeno37** [comment]
Bit of a necropost, but I just wanted to mention I came here from Google and I resolved the issue by disconnecting my Tailscale VPN.

**Comment by ooliver1** [comment]
@davidw-valve this is still an issue, it is reporting `127.0.1.1` as my public ip on linux but works perfectly fine with the suggested manual stuff with port forwarding
